File: typescript/base-hmi-types/src/screens/controls/HmiDataGridControl.ts
import { HmiControlWindowBase } from "../base/HmiControlWindowBase.js";
import { HmiColor } from "../base/HmiColor.js";
import { HmiObjectType } from "../base/HmiObjectType.js";
import { HmiProperty } from "../base/HmiProperty.js";
import { HmiDataGridColumn } from "./HmiDataGridColumn.js";
import { HmiDataGridDataSourceKind } from "./HmiDataGridDataSourceKind.js";
import { HmiDataGridSortDirection } from "./HmiDataGridSortDirection.js";

export class HmiDataGridControl extends HmiControlWindowBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiDataGridControl;
  }

  showExportCsv?: HmiProperty<boolean>;
  showProperties?: HmiProperty<boolean>;
  showStatusBar?: HmiProperty<boolean>;
  showToolbar?: HmiProperty<boolean>;
  toolbarIconSize?: HmiProperty<string>;
  dataSourceKind?: HmiProperty<HmiDataGridDataSourceKind>;
  sourceDataSourceKind?: string;
  dataSourceName?: HmiProperty<string>;
  tableOrView?: HmiProperty<string>;
  timeSortDirection?: HmiProperty<HmiDataGridSortDirection>;
  sourceTimeSortDirection?: string;
  historianInterpolatedMode?: HmiProperty<boolean>;
  historianInterpolationInterval?: HmiProperty<string>;
  maximumRows?: HmiProperty<number>;
  useLocalMachineTimeZone?: HmiProperty<boolean>;
  alternatingRowBackgroundColor?: HmiProperty<HmiColor>;
  gridLineColor?: HmiProperty<HmiColor>;
  headerRowHeightAutomatic?: HmiProperty<boolean>;
  headerRowHeight?: HmiProperty<number>;
  rowHeightAutomatic?: HmiProperty<boolean>;
  rowHeight?: HmiProperty<number>;
  readonly columnDefinitions: HmiDataGridColumn[] = [];
  timePeriodAbsoluteMode?: HmiProperty<boolean>;
  timePeriodDuration?: HmiProperty<string>;
  timePeriodStart?: HmiProperty<string>;
  timePeriodEnd?: HmiProperty<string>;
}

File: src/BaseHmiTypes/Screens/Controls/HmiDataGridControl.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public class HmiDataGridControl : HmiControlWindowBase
{
    public HmiDataGridControl()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiDataGridControl;
    }

    public HmiProperty<bool>? ShowExportCsv { get; set; }

    public HmiProperty<bool>? ShowProperties { get; set; }

    public HmiProperty<bool>? ShowStatusBar { get; set; }

    public HmiProperty<bool>? ShowToolbar { get; set; }

    public HmiProperty<string>? ToolbarIconSize { get; set; }

    public HmiProperty<int>? MaximumRows { get; set; }

    public HmiProperty<bool>? UseLocalMachineTimeZone { get; set; }

    public HmiProperty<HmiColor>? AlternatingRowBackgroundColor { get; set; }

    public HmiProperty<HmiColor>? GridLineColor { get; set; }

    public HmiProperty<bool>? HeaderRowHeightAutomatic { get; set; }

    public HmiProperty<double>? HeaderRowHeight { get; set; }

    public HmiProperty<bool>? RowHeightAutomatic { get; set; }

    public HmiProperty<double>? RowHeight { get; set; }

    public IList<HmiDataGridColumn> ColumnDefinitions { get; } = new List<HmiDataGridColumn>();

    public HmiProperty<bool>? TimePeriodAbsoluteMode { get; set; }

    public HmiProperty<string>? TimePeriodDuration { get; set; }

    public HmiProperty<string>? TimePeriodStart { get; set; }

    public HmiProperty<string>? TimePeriodEnd { get; set; }
}
